lib/view_model/dashboard/dashboard_view_model.dart

```diff
@@ -536,6 +537,88 @@ abstract class DashboardViewModelBase with Store {
     return resp;
   }
 
+  @observable
+  late bool backgroundSyncNotificationsEnabled = sharedPreferences.getBool(PreferencesKey.backgroundSyncNotificationsEnabled) ?? false;
+
+  @action
+  Future<void> setBackgroundSyncNotificationsEnabled(bool value) async {
+    if (!value) {
+      backgroundSyncNotificationsEnabled = false;
+      sharedPreferences.setBool(PreferencesKey.backgroundSyncNotificationsEnabled, false);
+      return;
+    }
+    PermissionStatus permissionStatus = await Permission.notification.status;
+    if (permissionStatus == PermissionStatus.denied) {
+      final resp = await Permission.notification.request();
+      if (resp == PermissionStatus.denied) {
+        throw Exception("Notification permission denied");
+      }
+    }
+    backgroundSyncNotificationsEnabled = value;
+    await sharedPreferences.setBool(PreferencesKey.backgroundSyncNotificationsEnabled, value);
+  }
+
+
+  bool get hasBgsyncNetworkConstraints => Platform.isAndroid;
+  bool get hasBgsyncBatteryNotLowConstraints => Platform.isAndroid;
+  bool get hasBgsyncChargingConstraints => Platform.isAndroid;
+  bool get hasBgsyncDeviceIdleConstraints => Platform.isAndroid;
+
+  @observable
+  bool backgroundSyncNetworkUnmetered = false;
+
+  @observable
+  bool backgroundSyncBatteryNotLow = false;
+
+  @observable
+  bool backgroundSyncCharging = false;
+
+  @observable
+  bool backgroundSyncDeviceIdle = false;
+  
+  Future<void> _loadConstraints() async {
+    backgroundSyncNetworkUnmetered = await FlutterDaemon().getNetworkType();
+    backgroundSyncBatteryNotLow = await FlutterDaemon().getBatteryNotLow();
+    backgroundSyncCharging = await FlutterDaemon().getRequiresCharging();
+    backgroundSyncDeviceIdle = await FlutterDaemon().getDeviceIdle();
+  }
+
+  @action
+  Future<void> setBackgroundSyncNetworkUnmetered(bool value) async {
+    backgroundSyncNetworkUnmetered = value;
+    await FlutterDaemon().setNetworkType(value);
+    if (await isBackgroundSyncEnabled()) {
+      await enableBackgroundSync();
```

Comment:
is this named incorrectly, or why are we enabling it if it's already enabled?

Reply:
we are re-enabling it to make the new changes persist in android work manager, and we are doing that only when background sync is enabled to not start it if it was currently off.